source/DomainTheory/Taboos/ClassicalLiftingOfNaturalNumbers.lagda

```diff
@@ -0,0 +1,232 @@
+Tom de Jong, 22 & 23 June 2024
+
+We formalize what we've known for a long time: if the poset pictured below
+
+   0   1   2   3   ...
+     \ \   / /
+      \ \ / /
+         ⊥
+
+is ω-complete/directed complete, then Bishop's LPO holds.
+
+\begin{code}
+
+{-# OPTIONS --safe --without-K #-}
+
+open import MLTT.Spartan
+
+open import UF.FunExt
+open import UF.PropTrunc
+
+module DomainTheory.Taboos.ClassicalLiftingOfNaturalNumbers
+        (pt : propositional-truncations-exist)
+        (fe : Fun-Ext)
+       where
+
+open PropositionalTruncation pt
+
+open import DomainTheory.Basics.Dcpo pt fe 𝓤₀
+open import DomainTheory.Basics.Miscelanea pt fe 𝓤₀
+
+open import CoNaturals.Type renaming (ℕ∞-to-ℕ→𝟚 to ε)
+open import MLTT.Two-Properties
+open import MLTT.Plus-Properties
+open import Notation.CanonicalMap
+open import Taboos.LPO (λ 𝓤 𝓥 → fe)
+
+\end{code}
+
+The poset is defined as follows. Notice that the set of natural numbers is
+ordered discretely.
+
+NB: In this file, and this file only, we denote this poset by ℕ⊥. In
+CoNaturals/Sharp this notation is reserved for the lifting of the natural
+numbers which constructively *does* yield a directed complete
+poset. Classically, the constructions are equivalent, as formalized in
+Lifting.Miscelanea-PropExt-FunExt.
+
```

Comment:
Good. Maybe you can add that this the case only classically, and that it is the purpose of this file to prove this.

Reply:
Let me know if you want me to merge this as it is, or if you would be happy to add the additional remark.